cli/internal/encoding/gitoutput/validators.go

```diff
@@ -0,0 +1,111 @@
+package gitoutput
+
+import "bytes"
+
+var _allowedObjectType = []byte("blob tree commit ")
+var _allowedStatusChars = []byte(" MTADRCU?!")
+
+func checkValid(fieldType field, value *[]byte) error {
```

Comment:
I didn't understand that slices were passed as a (pointer + size) and I was trying to avoid passing the contents of the slice. I had it both ways at different points in time and it made no measurable performance difference.